src/client/mds_client.cpp

```diff
@@ -934,7 +946,8 @@ LIBCURVE_ERROR MDSClient::GetOrAllocateSegment(bool allocate,
         int chunksNum = pfs.chunks_size();
         if (allocate && chunksNum <= 0) {
```

Comment:
if GetOrAllocateSegment get segment fail, and the reason is no enough space, the program may not go to here.
```
bool ChunkSegmentAllocatorImpl::AllocateChunkSegment(FileType type,
        SegmentSizeType segmentSize, ChunkSizeType chunkSize,
        offset_t offset, PageFileSegment *segment)  {
        ..........
        segment->set_chunksize(chunkSize);
        segment->set_segmentsize(segmentSize);
        segment->set_startoffset(offset);

        // allocate chunks
        uint32_t chunkNum = segmentSize/chunkSize;
        std::vector<CopysetIdInfo> copysets;
        if (!topologyChunkAllocator_->
                AllocateChunkRoundRobinInSingleLogicalPool(
                type, chunkNum, chunkSize, &copysets)) {
            LOG(ERROR) << "AllocateChunkRoundRobinInSingleLogicalPool error";
            return false;
        }
```

```
message PageFileSegment {
    required uint32 logicalPoolID = 1;
    required uint32 segmentSize = 3;
    required uint32 chunkSize = 4;
    required uint64 startOffset = 2;
    repeated  PageFileChunkInfo chunks = 5;
}
```

The logicalPoolID is required. The RPC may fail here
 ` if (cntl->Failed()) { `

Check this problem please.

Reply:
The filed `PageFileSegment` is cleared when alloc segment failed, so it will not trigger RPC `EREQUEST ` eror.